src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryTable.cs

```diff
@@ -493,6 +504,33 @@ private static bool IsValueTypeErrorOrBlank<T>(FormulaValue val)
             return val is T || val is BlankValue || val is ErrorValue;
         }
 
+        private static FormulaValue DistinctValueType(List<(DValue<RecordValue> row, FormulaValue distinctValue)> pairs, IRContext irContext)
+        {
+            var lookup = new HashSet<object>();
+            var result = new List<DValue<RecordValue>>();
+            var name = ((TableType)irContext.ResultType).SingleColumnFieldName;
+            var type = irContext.ResultType;
+
+            foreach (var (row, distinctValue) in pairs)
+            {
+                if (distinctValue is ErrorValue errorValue)
+                {
+                    return errorValue;
+                }
+
+                var key = distinctValue.ToObject();
```

Comment:
>distinctValue.ToObject(); [](http://example.com/codeflow?start=26&length=25)

Can we gaurante this is a primitive object?

Reply:
Yes, CheckTypes guarantees it.

Reply:
How is CheckTypes  guaranteeing that?

Reply:
I removed the check since PA supports Record for distinct.

Reply:
Added a check in implementation now.